### PR TITLE
C#: Fix false positive alert for shadowing on record types.

### PR DESCRIPTION
**Pull request by michaelnebel**
When defining a record type, the compiler generates a Deconstruct method on the class or struct representing the record type.
This method has out parameters with the same name as property accessors.
In this PR we add this exception to the rule that identifies bad practices related to shadowing.

### CODE REVIEW

**Comment by michaelnebel** [comment]
Fixes https://github.com/github/codeql/issues/7609

**Review by hvitved — APPROVED** [review]
Excellent!

File: csharp/ql/lib/change-notes/2022-01-18-local-shadows-member.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* Added exception for record types auto generated Deconstruct method.
```
Comment (hvitved) [inline csharp/ql/lib/change-notes/2022-01-18-local-shadows-member.md:4]: The change note should mention `cs/local-shadows-member`. So something like
```
The query `cs/local-shadows-member` no longer highlights parameters of `record` types.
```

**Review by hvitved — DISMISSED** [review]

**Comment by hvitved** [comment]
> Will need to rebase after #7605 is merged.

Why is that?

**Review by hvitved — APPROVED** [review]

**Pull request merged**